src/sql/src/plan/query.rs

```diff
@@ -1175,7 +1175,7 @@ pub fn plan_ctes(
             }
         }
         CteBlock::MutuallyRecursive(ctes) => {
-            qcx.scx.require_unsafe_mode("WITH MUTUALLY_RECURSIVE")?;
+            qcx.scx.require_feature(crate::WITH_MUTUALLY_RECURSIVE)?;
```

Comment:
Note that I have introduce a dedicated constant for the feature flag name under `src/sql/lib.rs`.

Ideally this should be referencing the `name()` of `static WITH_MUTUALLY_RECURSIVE: ServerVar<bool>` from `vars.rs`, but the `adapter` code is not available in `mz_sql`. I think in the long term we should try to move `vars.rs` to some crate that is further up the dependency graph.